src/scrypt.rs

```diff
@@ -32,9 +32,43 @@ pub fn derive_key(pass: &Password, salt: &[u8; 32], n: u32, p: u32, r: u32) -> R
     }
 
     let mut derived_key = vec![0u8; KEY_LENGTH];
-    let scrypt_params = ScryptParams::new(log_n, r, p);
-    scrypt(pass.as_bytes(), salt, &scrypt_params, &mut derived_key);
+    let scrypt_params = ScryptParams::new(log_n, r, p)?;
+    scrypt(pass.as_bytes(), salt, &scrypt_params, &mut derived_key)?;
     let derived_right_bits = &derived_key[0..KEY_LENGTH_AES];
     let derived_left_bits = &derived_key[KEY_LENGTH_AES..KEY_LENGTH];
     Ok((derived_right_bits.to_vec(), derived_left_bits.to_vec()))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::password::Password;
+
+    #[test]
+    fn scrypt_test() {
```

Comment:
How about it? I added a test to check whether the result is the same as the previous implementation.
Expected values, `right_bits` and `left_bits` were brought from the previous implementation. : )

Reply:
LGTM